File: Cell.cpp
#include "Cell.hpp"
#include <vector>
#include <iostream>

Cell::Cell(Vector position) : m_position(position), m_status(0), m_isCircled(false), m_polygon(0)
{
	std::cout << "Ячейка создана" << std::endl;
}

Vector Cell::GetPosition()
{
	return m_position;
}
int Cell::GetStatus()
{
	return m_status;
}
void Cell::SetStatus(int status)
{
	m_status = status;
}
bool Cell::GetIsCircled()
{
	return m_status;
}
void Cell::SetIsCircled(bool isCircled)
{
	m_isCircled = isCircled;
}
void Cell::SetPolygonSquare()
{
	std::cout << "Полигон сделан квадратом" << std::endl;
}
void Cell::SetPolygonTriangle(Vector direction)
{
	std::cout << "Полигон сделан треугольником" << std::endl;
}

File: lab5/java/Cell.java
public class Cell {
    
    private Vector position;
	private int status;
	private boolean isCircled;
	public Vector[] polygon;


	public Cell(Vector position)
    {
       position = new Vector(position.getX(), position.getY());
       setStatus(0);
       setIsCirecled(false);
       setPolygonSquare();
       System.out.println("Cell is created");
    }
	public Cell(int x, int y)
    {
       position = new Vector(x, y);
       setStatus(0);
       setIsCirecled(false);
       setPolygonSquare();
       System.out.println("Cell is created");
    }
    public Vector getPosition()
    {
        return new Vector(position.getX(), position.getY());
    }

    public int getStatus()
    {
        return status;
    }
    public void setStatus(int status)
    {
        this.status = status;
    }
    public boolean getIsCirecled()
    {
        return  isCircled;
    }
    public void setIsCirecled(boolean isCircled)
    {
        this.isCircled = isCircled;
    }
    public Vector[] getPolygon()
    {
        return polygon.clone();
    }
	public void setPolygonSquare()
    {
       polygon = new Vector[4];
    }
	public void setPolygonTriangle(Vector direction)
    {
       polygon = new Vector[3];
    }
}
